backend/src/main/java/com/hackner/musicompass/service/RecommendationService.java

```diff
@@ -48,4 +51,53 @@ public RecommendationTags changeRecommendationTags (RecommendationTagsDto recomm
         return recommendationCategoryMongoDb.findAll();
     }
 
+    public String getArtistRecommendation(RecommendationRequestDto recommendationRequestDto){
+
+        List<Artist> artists = artistMongoDb.findAll();
+
+        List<Artist> genreFilteredArtists = filterByRecommendationTags(artists, recommendationRequestDto.getGenres(), "genres");
+        List<Artist> genreRoleFilteredArtists = filterByRecommendationTags(genreFilteredArtists, recommendationRequestDto.getRoles(), "roles");
+        List<Artist> fullyFilteredArtists = filterByRecommendationTags(genreRoleFilteredArtists, recommendationRequestDto.getGender(), "gender");
+
+        if(fullyFilteredArtists.size() > 0){
+            Random rand = new Random();
+            return fullyFilteredArtists.get(rand.nextInt(fullyFilteredArtists.size())).getArtistName();
+        }
+
+        return "not-found";
+    }
+
+    public List<Artist> filterByRecommendationTags(List<Artist> artists, List<String> filterCriteria, String filterCategory ) {
```

Comment:
`filterCategory` could be an enum ;)

Reply:
I have rewritten my code to use an enum. but the real benefit of enums is not at all clear to me...

Reply:
In short: Compile time checks & documentation. When you declare a a parameter as String, every String is okay for the compiler, also enums clearly communicate which options are available